src/cli/opstooling-scripts.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# Using script location as pwd allows to utilize dependencies of opstooling-js,
+# namely tsnode and tsconfig-paths.
+# `readlink` is important to get around a symlink in node_modules/.bin,
+# so we would get actual location of the script
+script_dir="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
```

Comment:
The `readlink` command can fail and that's not being handled here. When that happens the command would turn into `dirname ""`, which outputs `.`. In that case `cd` would work since `cd .` always works, but the bug would go unnoticed.

A safer option would be to

```bash
set -eu -o pipefail

script_path="$(readlink -f "${BASH_SOURCE[0]}")"
script_dir="$(dirname "$script_path")"
```

With the `set -eu -o pipefail` bit it's not necessary to be checking for bad exit codes anywhere since those failures automatically exit the program; with that setup we focus on handling the expected failures in a few specific places, rather than everywhere.

```bash
# this command can fail
foo bar || :
```

All that being said, I think this could be fairly simply into a small Node.js script, which is not subject to Bash's weirdness while being more portable (less tools to depend on and is cross-platform).

```js
#!/usr/bin/env node

// fs.readLink can read the file's location from process.argv[0], but perhaps __dirname can work as well
// `cd` is not necessary because you can set the command's working directory with `spawnSync.options.cwd`

console.log(process.argv)
```

It's not a problem to put a shebang at the start, Node.js is fine with that.

Reply:
Yeah... both points make sense.
I'll see how js script would look like, and either go that direction or with `set -eu -o pipefail
`